### PR TITLE
Gwh apidocs 2152 euid update sdk csharp

### PR DESCRIPTION
**Pull request by genwhittTTD**
Updated with mods from compare with latest UID2 version, excluding sharing section.

### CODE REVIEW

File: api/v2/sdks/sdk-ref-csharp-dotnet.md [hunk 1]
```diff
@@ -45,53 +51,103 @@ The binary is published in this location:
...
 | `VersionNotSupported` |  The client library does not support the version of the encrypted token. |
 
+## Usage for DSPs
+
+The following instructions provide an example of how you can decode bid stream tokens using the EUID SDK for .NET as a DSP.
+
+1. Create a `BidstreamClient`:
+
+```cs
+var client = new BidstreamClient(UID2_BASE_URL, UID2_API_KEY, UID2_SECRET_KEY);
```
Comment (sunnywu) [inline api/v2/sdks/sdk-ref-csharp-dotnet.md:125]: should be all EUID instead of UID2

File: api/v2/sdks/sdk-ref-csharp-dotnet.md [hunk 2]
```diff
@@ -45,53 +51,103 @@ The binary is published in this location:
...
-| :--- | :--- | :--- |
-| `endpoint` | The endpoint for the EUID service. | N/A |
-| `authKey` | The authentication token that belongs to the client. For access to EUID, see [Contact Info](../getting-started/gs-account-setup.md#contact-info). | N/A |
+The `BidstreamClient` class allows you to decrypt EUID tokens into raw EUIDs.
 
-## Interface 
+For details on the bidding logic for handling user opt-outs, see [DSP Integration Guide](../guides/dsp-guide.md).
 
-The interface allows you to decrypt EUID advertising tokens and return the corresponding raw EUID. 
+The `SharingClient` class allows you to encrypt raw EUIDs into EUID tokens and decrypt EUID tokens into raw EUIDs.
```
Comment (sunnywu) [inline api/v2/sdks/sdk-ref-csharp-dotnet.md:70]: Shall we modify it like:
_The `SharingClient` class is not available for use in EUID and therefore can be ignored for now_

So we don't even mention about the functionalities not supported by EUID but at least clearly explicitly tell EUID participants to ignore it?

File: api/v2/sdks/sdk-ref-csharp-dotnet.md [hunk 3]
```diff
@@ -45,53 +51,103 @@ The binary is published in this location:
...
+| `KeysNotSynced` | The client has failed to synchronize keys from the EUID service. |
+| `KeyInactive` | The encryption key is not active. |
+| `EncryptionFailure` | A generic encryption failure occurred. |
+<!-- `TokenDecryptFailure` intentionally omitted. Does not seem to be used by SharingClient. -->
 
-### Response Content
+### Decryption Response Content
 
-Available information returned through the SDK is outlined in the following table.
+Whether decrypting with the `BidstreamClient` or the `SharingClient`, the SDK returns the following information:
```
Comment (sunnywu) [inline api/v2/sdks/sdk-ref-csharp-dotnet.md:98]: just remove `SharingClient`

File: api/v2/sdks/sdk-ref-csharp-dotnet.md [hunk 4]
```diff
@@ -45,53 +51,103 @@ The binary is published in this location:
...
-## Interface 
+For details on the bidding logic for handling user opt-outs, see [DSP Integration Guide](../guides/dsp-guide.md).
 
-The interface allows you to decrypt EUID advertising tokens and return the corresponding raw EUID. 
+The `SharingClient` class allows you to encrypt raw EUIDs into EUID tokens and decrypt EUID tokens into raw EUIDs.
 
 >NOTE: When you use an SDK, you do not need to store or manage decryption keys.
 
-If you're a DSP, for bidding, call the interface to decrypt an EUID advertising token and return the EUID. For details on the bidding logic for handling user opt-outs, see [DSP Integration Guide](../guides/dsp-guide.md).
+### Encryption Response Content
```
Comment (sunnywu) [inline api/v2/sdks/sdk-ref-csharp-dotnet.md:74]: Remove the whole Encryption Response Content and Encryption Response Statuses section as they are for Sharing only

**Comment by sunnywu** [comment]
Please don't merge and publish this until @jon8787 fixes some C# SDK issues, see slack for details

File: api/v2/sdks/sdk-ref-csharp-dotnet.md (hunk 2 above)
Comment (genwhittTTD) [reply api/v2/sdks/sdk-ref-csharp-dotnet.md:70]: @sunnywu I have a question on that. Basically, it depends whether it's visible to EUID people? AFAIK it's the same SDK and therefore, yes, if there is functionality there that isn't in use for EUID we should note it. Is that the case here?

File: api/v2/sdks/sdk-ref-csharp-dotnet.md (hunk 4 above)
Comment (genwhittTTD) [reply api/v2/sdks/sdk-ref-csharp-dotnet.md:74]: Done, thx.

File: api/v2/sdks/sdk-ref-csharp-dotnet.md (hunk 3 above)
Comment (genwhittTTD) [reply api/v2/sdks/sdk-ref-csharp-dotnet.md:98]: Done.

File: api/v2/sdks/sdk-ref-csharp-dotnet.md (hunk 2 above)
Comment (sunnywu) [reply api/v2/sdks/sdk-ref-csharp-dotnet.md]: Yes your assumption and result looks right!

File: api/v2/sdks/sdk-ref-csharp-dotnet.md [hunk 5]
```diff
@@ -45,53 +49,83 @@ The binary is published in this location:
...
-```
+- The `SharingClient` class is part of the SDK but is not currently available for use in EUID.
 
-### Response Content
+- When you use an SDK, you do not need to store or manage decryption keys.
 
-Available information returned through the SDK is outlined in the following table.
+### Decryption Response Content
+
+When decrypting with the `SharingClient`, the SDK returns the following information:
```
Comment (sunnywu) [inline api/v2/sdks/sdk-ref-csharp-dotnet.md:76]: Maybe let's change this SharingClient to BidstreamClient for now as the rest of contents still applicable

Comment (genwhittTTD) [reply api/v2/sdks/sdk-ref-csharp-dotnet.md:76]: Oops did I remove the wrong one? Fixed now. Thx much.

**Pull request merged**